### DISCUSSION TITLE
Warning on installation in new units

### DISCUSSION THREAD

**Discussion by eduardofilo** [Rparadise-Team/Koriki#9] [Announcements]
It has recently known that Miyoo Mini units manufactured since mid-December carry hardware (a new flash chip) and software changes (a couple of new firmware versions 202212160314 and 202212240121 have been located). This has affected the Onion community because of a script that was running during the installation of this system, already fixed, that made changes to the flash, which could brick the console. Koriki does not make such changes to the flash, so problems of this kind are not to be expected. However, there could be other kinds of problems, related for example to changes in versions of libraries contained in the firmware. There are also doubts about the system for saving settings (brightness, screen colors and audio fix), which seems to have changed being now redirected to the microSD. Due to these doubts we do not know if the changes introduced in the console by the manufacturer will affect the operation of Koriki. We insist, it is not expected that using Koriki on these new Miyoo Mini units will produce bricks, we only have doubts about the correct operation of the software included in Koriki. These doubts will only be cleared when someone tries to use Koriki in one of these new units. Although, as we say, the test does not seem dangerous, it would be advisable that it is carried out by someone with the ability to solve the problems that may arise. In the meantime we can only confirm, as stated in the [installation guide](https://github.com/Rparadise-Team/Koriki/wiki/En:-01.-Install-&-Upgrade), that Koriki has only been tested on Miyoo Mini v1 and v2 units with firmware 20220419. As more details become available we will update this information.

**Comment by KuDesNikdid** [comment]
It fault of onion installer script! Its fully save for korriki.

**Reply by KuDesNikdid** [reply]
They just comment garbage code and say "we fix chips and miyoo firmware")))
https://4pda.to/forum/index.php?showtopic=1049150&view=findpost&p=119824816

**Comment by eduardofilo** [comment]
We have had (remote) access to units with the new firmwares. The result of using Koriki in both versions is:

* 202212160314: System works (i.e. SimpleMenu starts and applications/emulators can be launched) but some settings changes (audiofix, brightness, screen color parameters) are not persisted and applied.
* 202212240121: SimpleMenu does not start, the screen remains black after the first bootscreens appear.

The problems of both versions are related to the same reason. It is that the path of the file where the system keeps the settings has changed. Before it was in `/appconfigs/system.json` and now it is in the root of the SD, i.e. `/mnt/SDCARD/system.json`.

In future versions of Koriki the location of the settings file will be managed automatically, but in the current version (1.0) it will be done explicitly, i.e. a different version will be used depending on the firmware of the machine on which we are installing. Thus, of the two files that are available for download in the repository [release](https://github.com/Rparadise-Team/Koriki/releases/tag/v1.0), the correspondence is as follows:

* `koriki_1.0.zip`: Miyoo Mini v1 and v2 with 20220419 firmware.
* `koriki_1.0_v3.zip`: Miyoo Mini v3 with 202212160314 or 202212240121 firmware.
